### DISCUSSION TITLE
Account Management Snaps: Support non-EVM address

### DISCUSSION THREAD

**Discussion by AnthonyLaw** [MetaMask/snaps#2252] [Ideas]
## Introduce
Using Keyring to create an account and the account shown in metamask is cool stuff.

I was thinking if it allows adding non-evm addresses in the metamask wallet even awesome.

## Advantage
- If creating non-evm account can be directly handled in metamask, instead of a custom page, it will speed up non-evm account implementation.
- For users, easy for them to manage accounts, all account (snap) is shown in metamask.

## Summary
Blockchain: Symbol
Refer repo: https://github.com/MetaMask/snap-simple-keyring

I was exploring keyring with non-evm accounts, and I successfully created the non-evm account, but there was an error.

when this event is triggered.
`await this.#emitEvent(KeyringEvent.AccountCreated, { account });`

<img width="761" alt="Screenshot 2024-03-05 at 11 47 17 AM" src="https://github.com/MetaMask/snaps/assets/5649156/49703adf-b844-48df-bc7c-d34032d8bf54">

**Comment by danroc** [comment]
Hey @AnthonyLaw, we are indeed considering supporting non-EVM accounts!

The Keyring API was made with this feature in mind, but we need to make some internal changes to MetaMask first. Essentially, we need to support non-EVM addresses in the UI and be able to interact with their chains to get balances and broadcast transactions.

TL;DR: That’s a really cool idea, we will think about it!
